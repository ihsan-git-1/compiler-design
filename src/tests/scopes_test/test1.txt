class test {

    void main(){

        if(true){
            String a = "String";
        }
    }
    void a(){

        while(true){
            int y = 1;
        }
    }

    void l(){

    }


 }