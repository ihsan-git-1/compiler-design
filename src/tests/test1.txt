class MyHomePage extends StatefulWidget {
  State<MyHomePage> createState() => MyHomePageState();
}
class MyHomePageState extends State<MyHomePage> {

  Widget build(BuildContext context) {
  return Scaffold(
  body: Padding(
        padding: EdgeInsets.all(8),
        child: Row(
            children: [
              Expanded(
                child: Column(
                  children: [
                    Container(
                      height: 300,
                      child: Column(
                        children:  [
                          Text("Hi"),
                          Expanded(child: Image.network("../../images/test_image.jpg"))
                        ],
                      ),
                    )],
                ),
              ),
              Expanded(
                child: Column(
                  children: [
                    Expanded(
                      child:Container(
                        height: 200,
                        child: MaterialButton(
                        onPressed:(){
                            Get.toNamed("../../images/test_image.jpg");
                         },
                          child: Column(
                            children:  [
                              Text("Test 1"),
                              Expanded(child: Image.network("../../images/test_image2.jpg"))
                            ],
                          ),
                        ),
                      ) ,
                    ) ,
                    Expanded(
                      child:Container(
                        width: 200,

                        height: 200,
                        child: MaterialButton(
                        onPressed:(){
                        Get.toNamed("../../images/test_image2.jpg");
                        },

                          child: Column(
                            children:  [
                              Text("Test 2"),
                              Expanded(child: Image.network("../../images/test_image3.jpg"))
                            ],
                          ),
                        ),
                      ) ,
                    ) ,
                    Expanded(
                      child:Container(
                        width: 200,

                        height: 200,
                        child: MaterialButton(
                       onPressed:(){

                        },
                          child: Column(
                            children:  [
                              Text("Test 3"),
                              Expanded(child: Image.network("../../images/test_image.jpg"))
                            ],
                          ),
                        ),
                      ) ,
                    ) ,
                    Expanded(
                      child:Container(
                        width: 200,

                        height: 200,
                        child: MaterialButton(
                        onPressed:(){

                                                },

                          child: Column(
                            children:  [
                              Text("Test 4"),
                              Expanded(child: Image.network("../../images/test_image2.jpg"))
                            ],
                          ),
                        ),
                      ) ,
                    ) ,
                  ],
                ),
              ),
            ],
          ),
      ),
  );
  }

}

