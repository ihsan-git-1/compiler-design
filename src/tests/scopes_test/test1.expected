
*********** AST ************

Line : (1) Top Tree Declaration  Child Count =  1 Type = Top Tree
Line : (1) Class Declaration Name is test parent AllClassesDeclaration Child Count =  7 Type = class
Line : (3) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (3) Function name: main functionType: void parameters: null functionBody: Line : (3) FunctionBody{statementList=Line : (5) If Statement:  Condition: Line : (5) ConditionExpr{ value=true col=0 type='boolean' childCount=0 parent='null'} block: Line : (6) Dart Variable Declaration:  parent: Statement Child Count =  1 Type = object
Line : (6) String Declaration Line: Number of declarations on this line: 1Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type string)
 elseIfBlock:  elseBlock: null  parent: 'null'} col=0 type='Function' childCount=0 parent='null'} parent: 'null Child Count =  5 Type = function
 Line : (9) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (9) Function name: a functionType: void parameters: null functionBody: Line : (9) FunctionBody{statementList=Line : (11) WhileStatement: condition: Line : (11) ConditionExpr{ value=true col=0 type='boolean' childCount=0 parent='null'} block: Line : (12) Dart Variable Declaration:  parent: Statement Child Count =  1 Type = object
Line : (12) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer)
 parent: 'null Child Count:  5 Type:  condition col=0 type='Function' childCount=0 parent='null'} parent: 'null Child Count =  5 Type = function
 Line : (16) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (16) Function name: l functionType: void parameters: null functionBody: Empty Body parent: 'null Child Count =  5 Type = function ,  

*********** SYMBOL TABLE ************

Global Scope (0)
Class Scope (1) Parent Is Global Scope (0)
Function Scope main (2) Parent Is Class Scope (1)
If Scope (3) Parent Is Function Scope main (2)
Function Scope a (2) Parent Is Class Scope (1)
While Scope (3) Parent Is Function Scope a (2)
Function Scope l (2) Parent Is Class Scope (1)

Identifier a, Type string, Value : "String" , Scope If Scope (3)
Identifier y, Type int, Value : 1 , Scope While Scope (3)
